xcresult/tests/xcresult.rs

```diff
@@ -1,40 +1,44 @@
 use context::repo::RepoUrlParts;
-use ctor::ctor;
 use flate2::read::GzDecoder;
 use lazy_static::lazy_static;
-use std::fs::File;
+use std::{fs::File, path::Path};
 use tar::Archive;
 use temp_testdir::TempDir;
 use xcresult::XCResult;
 
 const ORG_URL_SLUG: &str = "trunk";
 
+fn unpack_archive_to_temp_dir<T: AsRef<Path>>(archive_file_path: T) -> TempDir {
+    let file = File::open(archive_file_path).unwrap();
+    let decoder = GzDecoder::new(file);
+    let mut archive = Archive::new(decoder);
+    let temp_dir = TempDir::default();
+    if let Err(e) = archive.unpack(temp_dir.as_ref()) {
+        panic!("failed to unpack data.tar.gz: {}", e);
+    }
+    temp_dir
+}
+
 lazy_static! {
-    static ref TEMP_DIR: TempDir = TempDir::default();
+    static ref TEMP_DIR_TEST_1: TempDir =
+        unpack_archive_to_temp_dir("tests/data/test1.xcresult.tar.gz");
+    static ref TEMP_DIR_TEST_3: TempDir =
+        unpack_archive_to_temp_dir("tests/data/test3.xcresult.tar.gz");
+    static ref TEMP_DIR_TEST_4: TempDir =
+        unpack_archive_to_temp_dir("tests/data/test4.xcresult.tar.gz");
+    static ref TEMP_DIR_TEST_EXPECTED_FAILURES: TempDir =
+        unpack_archive_to_temp_dir("tests/data/test-ExpectedFailures.xcresult.tar.gz");
     static ref REPO: RepoUrlParts = RepoUrlParts {
         host: "github.com".to_string(),
         owner: "trunk-io".to_string(),
         name: "analytics-cli".to_string()
     };
 }
 
-#[cfg(test)]
-#[ctor]
-fn init() {
-    let path = "tests/data.tar.gz";
-    let file = File::open(path).unwrap();
-    let decoder = GzDecoder::new(file);
-    let mut archive = Archive::new(decoder);
-    match archive.unpack(TEMP_DIR.as_ref()) {
-        Ok(_) => (),
-        Err(e) => panic!("failed to unpack data.tar.gz: {}", e),
-    }
-}
```

Comment:
I don't think `ctor` gets us any more functionality than what `lazy_static` already allows, which is to lazily initialize anything and stick it in a static reference. Let me know if this is off and there's something more I'm not seeing.